### PR TITLE
Build storage from scratch to avoid corrupting it

### PR DESCRIPTION
**Pull request by Adamantios**
Ensure `trader-quickstart` is robust against small errors during the first run and doesn't need to recur to crash and ask the user to manually investigate the issue.

### CODE REVIEW

File: run_service.sh [hunk 1]
```diff
@@ -175,6 +175,92 @@ get_on_chain_service_state() {
...
+# Function to create the .trader_runner storage
+create_storage() {
+    echo "This is the first run of the script. The script will generate new operator and agent instance addresses."
+    echo ""
+    # Generate the operator's key
+    address_start_position=17
+    pkey_start_position=21
+    poetry run autonomy generate-key -n1 ethereum
+    mv "$keys_json" "../$keys_json_path"
+    operator_address=$(sed -n 3p "../$keys_json_path")
```
Comment (jmoreira-valory) [inline run_service.sh:197]: I created a couple of functions to extract private key and address from keys.json files. Try using them.

File: run_service.sh [hunk 2]
```diff
@@ -175,6 +175,92 @@ get_on_chain_service_state() {
...
+operator_keys_file="$store/operator_keys.json"
+keys_json="keys.json"
+keys_json_path="$store/$keys_json"
+agent_address_path="$store/agent_address.txt"
+service_id_path="$store/service_id.txt"
+service_safe_address_path="$store/service_safe_address.txt"
+store_readme_path="$store/README.txt"
+
+# Function to create the .trader_runner storage
+create_storage() {
```
Comment (jmoreira-valory) [inline run_service.sh:189]: Where is rpc stored in this function?

File: run_service.sh [hunk 3]
```diff
@@ -175,6 +175,92 @@ get_on_chain_service_state() {
...
+store=".trader_runner"
+rpc_path="$store/rpc.txt"
+operator_keys_file="$store/operator_keys.json"
+keys_json="keys.json"
+keys_json_path="$store/$keys_json"
+agent_address_path="$store/agent_address.txt"
+service_id_path="$store/service_id.txt"
+service_safe_address_path="$store/service_safe_address.txt"
+store_readme_path="$store/README.txt"
+
```
Comment (jmoreira-valory) [inline run_service.sh:187]: You should also define 2 files for the agent and operator key in standalone format (it's redundant, but it's used by several commands)

File: run_service.sh (hunk 2 above)
Comment (Adamantios) [reply run_service.sh:189]: It is stored outside the function:

https://github.com/valory-xyz/trader-quickstart/blob/0f99c967c2e7e8107c60b35f37b161a14fb5dce1/run_service.sh#L336

We do not want to clone the repo if the RPC does not support `eth_newFilter`.

File: run_service.sh (hunk 1 above)
Comment (Adamantios) [reply run_service.sh:197]: This should be done on a separate PR. This one just moves existing stuff in the new method, the suggested refactoring should be performed separately.

**Review by jmoreira-valory — APPROVED** [review]

**Pull request merged**